library/src/compute/calc.rs

```diff
@@ -404,6 +408,170 @@ pub fn log(
     Value::Float(result)
 }
 
+/// Calculates the product of a range of numbers. Used to calculate permutations.
```

Comment:
could you inverse the order so that the helper function comes after the main function?